### PR TITLE
Webdriver library

### PR DESCRIPTION
**Pull request by smangarole**
Tested the jenkins job. Looks good.

### CODE REVIEW

File: allure-generate.bat [hunk 1]
```diff
@@ -0,0 +1,6 @@
+if %CI% != true (
```
Comment (samirans89) [inline allure-generate.bat]: is this file needed in GitHub or is auto generated when the project is built?

File: allure-generate.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+if [ "$CI" != true ] ; then
```
Comment (samirans89) [inline allure-generate.sh]: is this file needed in GitHub or is auto generated when the project is built?

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+repositories {
+    mavenLocal()
+    maven {
+        url = uri('https://repo.maven.apache.org/maven2/')
+    }
+    maven {
+        url =uri('https://plugins.gradle.org/m2/')
+    }
+}
+dependencies {
```
Comment (samirans89) [inline build.gradle:16]: webdriver lib?

File: src/main/java/com/browsersatck/examples/pages/Checkout.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    private static final String ADDRESS_LINE1_ID = "addressLine1Input";
+    private static final String PROVINCE_ID = "provinceInput";
+    private static final String POST_CODE_ID = "postCodeInput";
+    private static final String SUBMIT_BUTTON_ID = "checkout-shipping-continue";
+
+    private static final String FIRST_NAME = "Raveendra";
+    private static final String LAST_NAME = "Tudangil";
+    private static final String ADDRESS_LINE1 = "Mumbai";
+    private static final String PROVINCE = "Maharashtra";
+    private static final String POST_CODE = "400068";
```
Comment (samirans89) [inline src/main/java/com/browsersatck/examples/pages/Checkout.java]: anonymise names and addresses.

File: src/main/java/com/browsersatck/examples/pages/Login.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package com.browsersatck.examples.pages;
+
+import com.browsersatck.examples.utils.LoggedInNavBarComponent;
+import com.browsersatck.examples.utils.UserCredentialUtil;
+import com.browsersatck.examples.utils.WebDriverWaitUtil;
```
Comment (samirans89) [inline src/main/java/com/browsersatck/examples/pages/Login.java]: correct package name spell error for browserstack.

File: src/main/java/com/browsersatck/examples/pages/Login.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+
+public class Login extends NavigablePage {
+
+    private static final String USER_INPUT_XPATH = "//input[@id='react-select-2-input']";
```
Comment (samirans89) [inline src/main/java/com/browsersatck/examples/pages/Login.java]: replace with CSS

input[id='react-select-2-input']

File: src/main/java/com/browsersatck/examples/pages/Login.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    private static final String PASSWORD_INPUT_XPATH = "//*[@id=\"react-select-3-input\"]";
```
Comment (samirans89) [inline src/main/java/com/browsersatck/examples/pages/Login.java]: replace with CSS
[id='react-select-3-input']

File: src/main/java/com/browsersatck/examples/pages/Offers.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package com.browsersatck.examples.pages;
+
+public class Offers {
+    public Offers() {
+        throw new UnsupportedOperationException("Method is unimplemented.");
```
Comment (samirans89) [inline src/main/java/com/browsersatck/examples/pages/Offers.java]: do we want to keep this open and unimplemented?

File: src/main/java/com/browsersatck/examples/pages/Orders.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
+package com.browsersatck.examples.pages;
+
+import io.qameta.allure.Step;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class Orders extends NavigablePage {
+
+    private static final String ORDERS_ID = "orders";
+    private static final String PRODUCT_COST_XPATH = "//span[@class='a-size-small a-color-price']";
```
Comment (samirans89) [inline src/main/java/com/browsersatck/examples/pages/Orders.java]: Replace with CSS:
span[class='a-size-small a-color-price']

File: src/main/java/com/browsersatck/examples/utils/LoggedInNavBarComponent.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
+package com.browsersatck.examples.utils;
+
+import com.browsersatck.examples.pages.Orders;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class LoggedInNavBarComponent extends NavBarComponent {
+
+    private static final String ORDERS_BUTTON_ID = "orders";
+    private static final String PRODUCT_COST_XPATH = "//span[@class='a-size-small a-color-price']";
```
Comment (samirans89) [inline src/main/java/com/browsersatck/examples/utils/LoggedInNavBarComponent.java]: replace with CSS

span[class='a-size-small a-color-price']

File: src/main/java/com/browsersatck/examples/utils/NavBarComponent.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+import com.browsersatck.examples.pages.Login;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public abstract class NavBarComponent {
+
+    private static final String SIGN_IN_BUTTON_ID = "signin";
+    private static final String SIGN_OUT_BUTTON_ID = "logout";
+    private static final String USER_INPUT_XPATH = "//input[@id='react-select-2-input']";
```
Comment (samirans89) [inline src/main/java/com/browsersatck/examples/utils/NavBarComponent.java]: replace with CSS

input[id='react-select-2-input']

File: src/test/java/com/browserstack/examples/helpers/Constants.java [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+        public static final String FAVOURITES_COUNT = "Favourites should not be empty after marking items favourite.";
+        public static final String OFFERS_PAGE_NOT_LOADED_ON_TIME = "Offers page is taking too long to load.";
+        public static final String OFFER_MESSAGES_NOT_FOUND = "No offer messages found.";
+        public static final String NO_OFFERS_LOADED = "No Offers loaded.";
+        public static final String CONFIRMATION_FAILED = "Confirmation Failed";
+    }
+
+    public static class ElementLocators {
+        public static final String HOME_PAGE_TITLE = "StackDemo";
+        public static final String RELOAD_SPINNER_XPATH = "//div[@class='spinner lds-ring']";
```
Comment (samirans89) [inline src/test/java/com/browserstack/examples/helpers/Constants.java]: replace with CSS

div[class='spinner lds-ring']

File: src/test/java/com/browserstack/examples/helpers/Constants.java [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+        public static final String PRODUCT_PRICE_VALUE_CARD_CLASS = "val";
+        public static final String PRODUCT_COST_BOLD_TAG = "b";
+        public static final String APPLE_FILTER_XPATH = "//div[@class='filters']//span[text()='Apple']";
+        public static final String SAMSUNG_FILTER_XPATH = "//div[@class='filters']//span[text()='Samsung']";
+        public static final String FAVOURITES_BUTTON_ID = "favourites";
+        public static final String API_ERROR_CLASS = "api-error";
+        public static final String PRODUCT_IMAGE_TAG = "img";
+        public static final String PRODUCT_IMAGE_SOURCE_ATTRIBUTE = "src";
+        public static final String ORDERS_BUTTON_ID = "orders";
+        public static final String PRODUCT_COST_XPATH = "//span[@class='a-size-small a-color-price']";
```
Comment (samirans89) [inline src/test/java/com/browserstack/examples/helpers/Constants.java]: Replace with CSS.

span[class='a-size-small a-color-price']

File: src/test/java/com/browserstack/examples/helpers/Constants.java [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+        public static final String FAVOURITE_BUTTON_XPATH = "//div[@class='shelf-stopper']";
```
Comment (samirans89) [inline src/test/java/com/browserstack/examples/helpers/Constants.java]: Replace with CSS

div[class='shelf-stopper']

File: src/test/java/com/browserstack/examples/helpers/Constants.java [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+        public static final String FAVOURITE_BUTTON_CLICKED_XPATH = "//button[@class='MuiButtonBase-root MuiIconButton-root Button clicked ']";
```
Comment (samirans89) [inline src/test/java/com/browserstack/examples/helpers/Constants.java]: Replace with CSS
button[class='MuiButtonBase-root MuiIconButton-root Button clicked ']

**Review by samirans89 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (raveerocks) [reply build.gradle:16]: Added for allure plugin

**Review by samiran-bs — APPROVED** [review]

**Pull request merged**